Remove `enable_proposal_linking` setting

greenwoodt opened the pull request:
#### :tophat: What? Why?
Removal of unnecessary setting `enable_proposal_linking`, linking **Proposals** to other modules: **Accountability**, **Budgets** and **Meetings**. 

The rhetoric reasoning of this removal is due to extending and improving the settings usage with proposed features such as: #13067 & #14289. 

The current setting is also arguably against the Traceability principle of the Social Contract with regards to the usage of the software. 

#### :pushpin: Related Issues
- Fixes #14415 

#### Testing

### :camera: Screenshots

:hearts: Thank you!

alecslupu commented on decidim-accountability/app/views/decidim/accountability/admin/results/_form.html.erb:
```diff
@@ -57,11 +57,9 @@
         </div>
       </div>
 
-    <% if Decidim::Accountability.enable_proposal_linking %>
```
If you remove this, and the proposals module is not installed / enabled, this will actually throw an error as the partial path could not be found.

alecslupu replied:
How about replacing that `if ` with something like `if Decidim.module_installed?(:proposals)` ...

greenwoodt replied:
How does this method work? @alecslupu in the edit of each module concerned, the option to pick a proposal in the form still exists? 

![image](https://github.com/user-attachments/assets/9ba4b169-b1cb-4eb1-835b-d97269adba86)

alecslupu commented on decidim-budgets/app/views/decidim/budgets/admin/projects/_form.html.erb:
```diff
@@ -27,11 +27,9 @@
         <% end %>
       <% end %>
 
-    <% if Decidim::Budgets.enable_proposal_linking %>
```
Same here

andreslucena commented on decidim-accountability/app/views/decidim/accountability/admin/results/_form.html.erb:
```diff
@@ -57,7 +57,7 @@
       </div>
 
-    <% if Decidim::Accountability.enable_proposal_linking %>
+    <% if Decidim.module_installed(:proposals) %>
```
```suggestion
    <% if Decidim.module_installed?(:proposals) %>
```

andreslucena commented on decidim-budgets/app/views/decidim/budgets/admin/projects/_form.html.erb:
```diff
@@ -27,7 +27,7 @@
       <% end %>
 
-    <% if Decidim::Budgets.enable_proposal_linking %>
+    <% if Decidim.module_installed(:proposals) %>
```
```suggestion
    <% if Decidim.module_installed?(:proposals) %>
```

andreslucena commented on decidim-meetings/app/views/decidim/meetings/meeting_closes/_form.html.erb:
```diff
@@ -2,6 +2,6 @@
 <%= form.number_field :attendees_count, min: 0, help_text: t("decidim.forms.meetings.attendees_count_help_text") %>
 
-<% if Decidim::Meetings.enable_proposal_linking %>
+<% if Decidim.module_installed(:proposals) %>
```
```suggestion
<% if Decidim.module_installed?(:proposals) %>
```

andreslucena reviewed (commented):
Still many references:

```
$ rg BUDGETS_ENABLE_PROPOSAL_LINKING .
./decidim-generators/lib/decidim/generators/test/generator_examples.rb
242:      "BUDGETS_ENABLE_PROPOSAL_LINKING" => "false",

./decidim-generators/lib/decidim/generators/app_templates/initializer.rb
467:#     unless Decidim::Env.new("BUDGETS_ENABLE_PROPOSAL_LINKING", "auto").default_or_present_if_exists.to_s == "auto"
468:#       config.enable_proposal_linking = Decidim::Env.new("BUDGETS_ENABLE_PROPOSAL_LINKING", "auto").default_or_present_if_exists

./docs/modules/configure/pages/environment_variables.adoc
735:|BUDGETS_ENABLE_PROPOSAL_LINKING
``` 

```
$ rg enable_proposal_linking
decidim-meetings/app/views/decidim/meetings/admin/meeting_closes/_form.html.erb
22:    <% if Decidim::Meetings.enable_proposal_linking %>

decidim-meetings/lib/decidim/api/meeting_type.rb
14:      implements Decidim::Meetings::LinkedResourcesInterface if Decidim::Meetings.enable_proposal_linking

decidim-meetings/lib/decidim/api/linked_resources_interface.rb
11:      if Decidim::Meetings.enable_proposal_linking

decidim-meetings/lib/decidim/meetings.rb
23:    config_accessor :enable_proposal_linking do
                                                                                                                                       decidim-meetings/app/models/decidim/meetings/meeting.rb
321:        return [] unless Decidim::Meetings.enable_proposal_linking
                                                                                                                                       decidim-meetings/app/presenters/decidim/meetings/meeting_presenter.rb
139:        return unless Decidim::Meetings.enable_proposal_linking
                                                                                                                                       decidim-meetings/spec/presenters/decidim/meetings/meeting_presenter_spec.rb
37:          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)

decidim-meetings/spec/system/admin/admin_manages_meetings_spec.rb
716:        allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)

decidim-generators/lib/decidim/generators/test/generator_examples.rb
353:      %w(decidim meetings enable_proposal_linking) => "auto",
447:      %w(decidim meetings enable_proposal_linking) => false,
639:      "enable_proposal_linking" => true,
647:      "enable_proposal_linking" => false,

decidim-generators/lib/decidim/generators/app_templates/initializer.rb
460:#       config.enable_proposal_linking = Decidim::Env.new("MEETINGS_ENABLE_PROPOSAL_LINKING", "auto").default_or_present_if_exists 468:#       config.enable_proposal_linking = Decidim::Env.new("BUDGETS_ENABLE_PROPOSAL_LINKING", "auto").default_or_present_if_exists
476:#       config.enable_proposal_linking = Decidim::Env.new("ACCOUNTABILITY_ENABLE_PROPOSAL_LINKING", "auto").default_or_present_if_exists

decidim-meetings/spec/models/meeting_spec.rb
472:          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)

decidim-accountability/spec/shared/manage_results_examples.rb
17:        allow(Decidim::Accountability).to receive(:enable_proposal_linking).and_return(false)
decidim-budgets/spec/shared/manage_projects_examples.rb                                                                                69:        allow(Decidim::Budgets).to receive(:enable_proposal_linking).and_return(false)

greenwoodt commented:
> Still many references:
> 
Thanks @andreslucena! I'll a request a review once the PR is ready!

greenwoodt commented:
@andreslucena ready again for another set of eyes FYI.

andreslucena reviewed (changes requested):
Can you check these out please? 
```
$ rg enable_proposal_linking
decidim-budgets/spec/shared/manage_projects_examples.rb
69:        allow(Decidim::Budgets).to receive(:enable_proposal_linking).and_return(false)

$ rg ENABLE_PROPOSAL_LINK
docs/modules/configure/pages/environment_variables.adoc
718:|MEETINGS_ENABLE_PROPOSAL_LINKING

```

greenwoodt commented:
> $ rg enable_proposal_linking
> decidim-budgets/spec/shared/manage_projects_examples.rb
> 69:        allow(Decidim::Budgets).to receive(:enable_proposal_linking).and_return(false)

Done @andreslucena! I don't however, see any reference to `:enable_proposal_linking` within decidim-budgets/spec/shared/manage_projects_examples.rb on this branch.

andreslucena commented:
> Done @andreslucena! I don't however, see any reference to :enable_proposal_linking within decidim-budgets/spec/shared/manage_projects_examples.rb on this branch.

Cannot find it anymore. Great job!

andreslucena commented on decidim-accountability/spec/shared/manage_results_examples.rb:
```diff
@@ -7,23 +7,6 @@
-
-    context "when proposal linking is disabled" do
-      before do
-        allow(Decidim::Accountability).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena commented on decidim-budgets/spec/shared/manage_projects_examples.rb:
```diff
@@ -63,19 +63,6 @@
-
-    context "when proposal linking is disabled" do
-      before do
-        allow(Decidim::Budgets).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena commented on decidim-meetings/spec/models/meeting_spec.rb:
```diff
@@ -466,17 +466,6 @@ module Decidim::Meetings
-
-      context "when proposal linking is disabled" do
-        before do
-          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena commented on decidim-meetings/spec/presenters/decidim/meetings/meeting_presenter_spec.rb:
```diff
@@ -31,17 +31,6 @@ module Decidim::Meetings
-
-      context "when proposal linking is disabled" do
-        before do
-          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena commented on decidim-meetings/spec/system/admin/admin_manages_meetings_spec.rb:
```diff
@@ -710,24 +710,6 @@
-
-    context "when proposal linking is disabled" do
-      before do
-        allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena commented on decidim-meetings/spec/system/user_close_meeting_spec.rb:
```diff
@@ -153,26 +153,6 @@
-
-    context "when proposal linking is disabled" do
-      before do
-        allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
```
This whole example should be left, changing this line with 

```
 allow(Decidim).to receive(:module_installed?).and_call_original
 ```

andreslucena reviewed (changes requested):
I think we should left the examples.

Apart of the change that i mention about the stub, you need to also change the example description (from "when proposal linking is disabled" to "when the proposal module is installed")

greenwoodt commented:
> I think we should left the examples.
> 
> Apart of the change that i mention about the stub, you need to also change the example description (from "when proposal linking is disabled" to "when the proposal module is installed")

@andreslucena I've reverted these and added the following method as per suggestion.

andreslucena commented:
> This setting will therefore be removed on all maintained versions of Decidim which include:

Just to clarify, the policy is to not remove this kind of settings in old versions, as this isn't a bugfix but a change/removal.

andreslucena commented on decidim-meetings/spec/system/user_close_meeting_spec.rb:
```diff
@@ -154,9 +154,9 @@
+    context "when the proposal module is installed" do
       before do
-        allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
+        allow(Decidim).to receive(:module_installed?).and_call_original
```
```suggestion
        allow(Decidim).to receive(:module_installed?).and_return(false)
```

andreslucena commented on decidim-meetings/spec/presenters/decidim/meetings/meeting_presenter_spec.rb:
```diff
@@ -32,9 +32,9 @@ module Decidim::Meetings
       end
 
-      context "when proposal linking is disabled" do
+      context "when the proposal module is installed" do
```
```suggestion
      context "when the proposal module is not installed" do
```

andreslucena commented on decidim-meetings/spec/presenters/decidim/meetings/meeting_presenter_spec.rb:
```diff
@@ -32,9 +32,9 @@ module Decidim::Meetings
+      context "when the proposal module is installed" do
         before do
-          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
+          allow(Decidim).to receive(:module_installed?).and_call_original
```
```suggestion
          allow(Decidim).to receive(:module_installed?).and_return(false)
```

andreslucena reviewed (commented):
In Meetings module specs, Proposals is enabled by default, so the picker is available always. We need to do the false explicilty. 

On another hand, as we're doing a negative spec (`expect(page).to have_no_content "Proposals"`) we should do also the positive one (`expect(page).to have_content "Proposals"`). This is because when you do a negative spec it's easy to do errors or just to say things that stop making sense in the future. 

`expect(page).to have_no_content "Supercalifragilisticexpialidocious"` will always pass, but we also should put an example for when `expect(page).to have_content "Supercalifragilisticexpialidocious"` also pass 😄 

We should do this in all the specs that we're doing instead of just doing the `and_call_original` method

andreslucena commented on decidim-meetings/spec/models/meeting_spec.rb:
```diff
@@ -467,9 +467,9 @@ module Decidim::Meetings
+      context "when the proposal module is installed" do
         before do
-          allow(Decidim::Meetings).to receive(:enable_proposal_linking).and_return(false)
+          allow(Decidim).to receive(:module_installed?).and_call_original
```
```suggestion
          allow(Decidim).to receive(:module_installed?).and_return(false)
```

andreslucena commented on decidim-accountability/spec/shared/manage_results_examples.rb:
```diff
@@ -7,22 +7,18 @@
-        allow(Decidim::Accountability).to receive(:enable_proposal_linking).and_return(false)
+  context "when the proposal module is installed" do
+    before do
+      allow(Decidim).to receive(:module_installed?).and_call_original
```
```suggestion
      allow(Decidim).to receive(:module_installed?).and_return(false)
```

andreslucena commented on decidim-budgets/spec/shared/manage_projects_examples.rb:
```diff
@@ -64,9 +64,9 @@
+    context "when the proposal module is installed" do
       before do
-        allow(Decidim::Budgets).to receive(:enable_proposal_linking).and_return(false)
+        allow(Decidim).to receive(:module_installed?).and_call_original
```
```suggestion
        allow(Decidim).to receive(:module_installed?).and_return(false)
```

andreslucena commented on decidim-meetings/spec/models/meeting_spec.rb:
```diff
@@ -467,9 +467,9 @@ module Decidim::Meetings
       end
 
-      context "when proposal linking is disabled" do
+      context "when the proposal module is installed" do
```
```suggestion
      context "when the proposal module is not installed" do
```

andreslucena commented on decidim-meetings/spec/system/admin/admin_manages_meetings_spec.rb:
```diff
@@ -711,9 +711,9 @@
     end
 
-    context "when proposal linking is disabled" do
+    context "when the proposal module is installed" do
```
```suggestion
    context "when the proposal module is not installed" do
```

andreslucena commented on decidim-meetings/spec/system/user_close_meeting_spec.rb:
```diff
@@ -154,9 +154,9 @@
     end
 
-    context "when proposal linking is disabled" do
+    context "when the proposal module is installed" do
```
```suggestion
    context "when the proposal module is not installed" do
```

andreslucena reviewed (dismissed):
![lgtm-looks-good-to-me](https://github.com/user-attachments/assets/86b72ad2-202a-4218-ae0b-08e243cfaa4d)

Passing the ball to @alecslupu

alecslupu reviewed (changes requested):
The PR looks ok, but considering we are changing the configuration API / (remove `enable_proposal_linking` config ), I would like to see a README entry that would help me while upgrading.

alecslupu commented on RELEASE_NOTES.md:
```diff
@@ -323,9 +323,13 @@ For ease of in operations, we also added the above command to the main `decidim:
+
+The rhetoric reasoning of this removal is due to extending and improving the settings usage with proposed features such as: [#13067] & [#14289].
+
+You can read more about this change on PR [#14453](https://github.com/decidim/decidim/pull/14453).
```
```suggestion
You can read more about this change on PR [#14453](https://github.com/decidim/decidim/pull/14453).

### 3.8. [[TITLE OF THE ACTION]]

You can read more about this change on PR [#XXXX](https://github.com/decidim/decidim/pull/XXXX).
```

alecslupu reviewed (approved):
:+1:
